### PR TITLE
Qwen3-30b learn to init

### PR DESCRIPTION
**Pull request by gagika**
# Description

Qwen3 MoE: Learn-to-Init distillation

Adds LTI soft distillation for Qwen3-30B-A3B-base MoE → custom student. Wires the LTI hook into `Qwen3MoeDecoderLayerToLinen`, adds warm-start init for the A/B bridges, and ships a recipe + setup doc.

# Tests

- End-to-end TPU v7x-8 run: ~12K steps, KL converges to ~0.41

gs://agagik-us/distillation/qwen3-30b-lti/qwen3-30b-lti-rope1e7-20260518-181330/checkpoints/

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

File: src/maxtext/layers/learn_to_init_layer.py [hunk 1]
```diff
@@ -33,12 +33,62 @@
 LTI_ORIGINAL_ATTENTION_PARAMS_NAME = "kernel"
 LTI_LAYER_PATH_PREFIXES = ("layers_", "dense_layers_", "moe_layers_")
 
+# Fallback teacher config for the LTI augment hook. The `_flat_config` dict
+# injection in train_distill is lost when HyperParameters is deep-copied during
+# nnx lazy_init, so the trainer also stashes the teacher config here.
+_TEACHER_CONFIG: Config | None = None
```
Comment (vlad-karp) [inline src/maxtext/layers/learn_to_init_layer.py:39]: The teacher config is available as an attribute of the student config, no need to have a global one.

File: src/maxtext/layers/learn_to_init_layer.py [hunk 2]
```diff
@@ -84,7 +134,12 @@ def _customize_attention_modules(config: Config, attn_module_name: str, module:
   target_names = LTI_MODIFIED_ATTENTION_PARAM_NAMES
 
   use_general_linear_map = config.lti_use_general_linear_map
-  teacher_config = config.teacher_config
+  teacher_config = getattr(config, "teacher_config", None) or _TEACHER_CONFIG
```
Comment (vlad-karp) [inline src/maxtext/layers/learn_to_init_layer.py:137]: do we really need _TEACHER_CONFIG?

File: src/maxtext/trainers/post_train/distillation/train_distill.py [hunk 3]
```diff
@@ -695,8 +695,11 @@ def train_distill(
       teacher_model = get_maxtext_model(teacher_config, mesh)
       teacher_model.eval()
 
-    # LTI phase needs the student initialization step to know about the teacher configuration
+    # LTI needs the teacher config at student lazy_init time. The dict
+    # injection is lost when HyperParameters is deep-copied, so also stash it
```
Comment (vlad-karp) [inline src/maxtext/trainers/post_train/distillation/train_distill.py:699]: can you explain when "The dict injection is lost when HyperParameters is deep-copie" happens - I didn't have this issue before.

**Review by JamesDeng42 — APPROVED** [review]
